### PR TITLE
fix/TRI 1149/cover all duplicated cases

### PR DESCRIPTION
**Pull request by ologbonowiwi**
- feat: select node by property key name
- feat: get `IfStatement`s recursively
- refactor: make `getNodeBody` more readable
- refactor: enshorter rule code

### CODE REVIEW

**Comment by ologbonowiwi** [comment]
This fixes #391

File: packages/eslint-plugin/lib/rules/no-duplicated-task-keys.js [hunk 1]
```diff
@@ -70,30 +74,48 @@ module.exports = {
...
+        const VariableDeclarations = body.filter((arg) => arg.type === 'VariableDeclaration');
+  
+        const grouped = groupVariableDeclarationsByTask(VariableDeclarations);
+  
+        const ExpressionStatements = body.filter((arg) => arg.type === 'ExpressionStatement');
+  
         // it'll be a map of taskName => [key1, key2, ...]
         const groupedByTask = groupExpressionsByTask(ExpressionStatements, grouped);
-
+
```
Comment (wesleymatosdev) [inline packages/eslint-plugin/lib/rules/no-duplicated-task-keys.js]:
```suggestion

```

File: packages/eslint-plugin/lib/rules/no-duplicated-task-keys.js [hunk 2]
```diff
@@ -70,30 +74,48 @@ module.exports = {
...
+        const ExpressionStatements = body.filter((arg) => arg.type === 'ExpressionStatement');
+  
         // it'll be a map of taskName => [key1, key2, ...]
         const groupedByTask = groupExpressionsByTask(ExpressionStatements, grouped);
-
+  
         groupedByTask.forEach((keys) => {
           const duplicated = keys.find((key, index) => keys.indexOf(key) !== index);
-
+
```
Comment (wesleymatosdev) [inline packages/eslint-plugin/lib/rules/no-duplicated-task-keys.js]:
```suggestion

```

**Comment by ericallam** [comment]
Thanks @ologbonowiwi for the fix! I was thinking, we should detect eslint usage and automatically configure this plugin during the `@trigger.dev/cli init` command. I'll create a separate issue for that though 👍

**Pull request merged**